components/net/http_loader.rs

```diff
@@ -120,6 +123,7 @@ impl Default for HttpState {
                 override_manager.clone(),
             )),
             override_manager,
+            embedder_proxy: Mutex::new(None),
```

Comment:
I needed to make proxy since this `default` implementation is used in tests and I couldn't find an easy way to create an embedder for the default implementation. Is that fine?

Reply:
I think we should remove this Default implementation from the engine if it's only used in tests, and make a test helper that can provide a meaningful value. Probably we should store a trait object, so that tests can do something useful with it.